### PR TITLE
Added price to huts

### PR DESCRIPTION
**Pull request by livborre**

### CODE REVIEW

File: src/huts/create/endpointSpec.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ describe("'create' endpoint for huts" , function() {
 
         endpoint = createEndpoint(hutsRepository);
 
-        try {
+        try {
```
Comment (hypesystem) [inline src/huts/create/endpointSpec.js]: Dangly space

File: src/huts/create/endpointSpec.js [hunk 2]
```diff
@@ -160,75 +160,75 @@ describe("'create' endpoint for huts" , function() {
...
+        } catch(error) {
+            actualError = error;
+        }
+
+        expect(actualError).toBe(null);
+        expect(res.status).toHaveBeenCalledWith(400);
+        expect(res.status).toHaveBeenCalledTimes(1);
+        expect(res.send).toHaveBeenCalledTimes(1);
+        expect(res.send).toHaveBeenCalledWith({requestErrors});
+    });
```
Comment (hypesystem) [inline src/huts/create/endpointSpec.js]: Why is this test commented out? It seems like it would make sense to test for this?

File: src/prices/repository.js [hunk 3]
```diff
@@ -86,13 +87,13 @@ async function createPrice(db, hutPrice) {
...
+                    ...priceData
                 }
             ]
         );
     } catch(error) {
         throw new Error("failed to insert a new price into 'prices' table");
     }
-
-    return;
+
```
Comment (hypesystem) [inline src/prices/repository.js]: Spaces on this line are odd

File: src/prices/repository.js [hunk 4]
```diff
@@ -52,22 +52,23 @@ async function findPrice(db, priceId) {
...
-        // sunday: result.rows[0].data.sunday
+        data: {
+            monday: result.rows[0].data.monday,
+            tuesday: result.rows[0].data.tuesday,
+            wednesday: result.rows[0].data.wednesday,
+            thursday: result.rows[0].data.thursday,
+            friday: result.rows[0].data.friday,
+            saturday: result.rows[0].data.saturday,
+            sunday: result.rows[0].data.sunday
+        }
```
Comment (hypesystem) [inline src/prices/repository.js]: There are two approaches to reducing the redundant code (`result.rows[0].data`) here...

1. [lodash pick](https://www.npmjs.com/package/lodash.pick) can be used to do `data: pick(result.rows[0].data, [ "monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday" ])`

2. Pick using destructuring before building the object: `let { monday, tuesday, wednesday, thursday, friday, saturday, sunday } = result.rows[0].data;` and then adding them in the dayPrices object as `data: { monday, tuesday, wednesday, thursday, friday, saturday, sunday }`.

Comment (hypesystem) [reply src/prices/repository.js]: Oops, just noticed that this is the prices repo that is removed again in next commit. But the feedback stands :smile:

File: src/prices/repository.js (hunk 3 above)
Comment (hypesystem) [reply src/prices/repository.js]: Oops, just noticed that this is the prices repo that is removed again in next commit. But the feedback stands :smile:

File: src/huts/create/endpointSpec.js [hunk 5]
```diff
@@ -159,75 +159,76 @@ describe("create endpoint" , function() {
...
+//         } catch(error) {
+//             actualError = error;
+//         }
+
+//         expect(actualError).toBe(null);
+//         expect(res.status).toHaveBeenCalledWith(400);
+//         expect(res.status).toHaveBeenCalledTimes(1);
+//         expect(res.send).toHaveBeenCalledTimes(1);
+//         expect(res.send).toHaveBeenCalledWith({requestErrors});
+//     });
```
Comment (hypesystem) [inline src/huts/create/endpointSpec.js]: Why are these tests commented out?

File: features/step_definitions/hutSteps.js [hunk 6]
```diff
@@ -11,21 +11,21 @@ When('I register a hut with the following information:', async function (dataTab
...
-    let pricesCleaned = prices.split(',').map(p => parseInt(p.trim()));
-    let pricesObject = {
-        monday: pricesCleaned[0],
-        tuesday: pricesCleaned[1],
-        wednesday: pricesCleaned[2],
-        thursday: pricesCleaned[3],
-        friday: pricesCleaned[4],
-        saturday: pricesCleaned[5],
-        sunday: pricesCleaned[6]
+    let dayPricesCleaned = dayPrices.split(',').map(p => parseInt(p.trim()));
```
Comment (hypesystem) [inline features/step_definitions/hutSteps.js]: You can get the variables like this, which provides a bit of context for why you are splitting the string:

```js
let [ monday, tuesday, wednesday, thursday, friday, saturday, sunday ] = dayPrices.split(',').map(p => parseInt(p.trim()));
```

Use it if you think it's better -- it's up to you :smile:

File: features/step_definitions/hutSteps.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 const { When, Then } = require("cucumber");
 const assert = require("assert");
+const {isValidUUID} = require("../../src/util.js");
```
Comment (hypesystem) [inline features/step_definitions/hutSteps.js]: I would like consistent spaces, so `{ isValidUUID }` :smile:

File: src/util.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+
+function isValidUUID(input) {
```
Comment (hypesystem) [inline src/util.js]: I think a better naming for this would probably be `isValidUuid`, seeing UUID as a single "word". That's how we generally consistently do it. Otherwise we'll end up with names like `XMLHttpRequest` (real example).

File: src/util.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+    let regex = new RegExp(/^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/, "i");
+    return regex.test(input);
+};
+
+module.exports = {
+    isValidUUID
+};
```
Comment (hypesystem) [inline src/util.js]: I generally think files like `util` are probably a bad idea: they are a messy collection of lots of unrelated code. I think a (slightly) better approach would be an `utils` folder with a file for each function (in this case, `utils/isValidUuid.js`).

File: src/huts/create/endpointSpec.js (hunk 2 above)
Comment (livborre) [reply src/huts/create/endpointSpec.js]: I don't see that it is commented out?

File: features/step_definitions/hutSteps.js (hunk 6 above)
Comment (livborre) [reply features/step_definitions/hutSteps.js]: much nicer :D

File: src/prices/repository.js (hunk 4 above)
Comment (livborre) [reply src/prices/repository.js]: very nice, thanks! I changed it in the priceRepo in another branch 😄

**Comment by livborre** [comment]
Thank you for the review! I believe i've corrected it all :)

File: src/huts/create/endpointSpec.js (hunk 2 above)
Comment (hypesystem) [reply src/huts/create/endpointSpec.js]: Odd! :smile:

File: src/huts/create/endpointSpec.js [hunk 10]
```diff
@@ -19,7 +19,7 @@ describe("'create' endpoint for huts" , function() {
         hutData = {
             hutName: "a hutname for testing puposes",
             street: "test street",
-            streetNumber: "1",
+            streetNumber: "1",
```
Comment (hypesystem) [inline src/huts/create/endpointSpec.js]: Dangly space why

Comment (hypesystem) [reply src/huts/create/endpointSpec.js]: If you make sure to run `git diff` before you add files, or `git diff --cached` before you commit, you can avoid committing little odd things like this :smile:

Comment (livborre) [reply src/huts/create/endpointSpec.js]: oh damn, sorry 😅
